assets/js/base/components/cart-checkout/address-form/address-form.tsx

```diff
@@ -214,6 +214,47 @@ const AddressForm = ( {
 					);
 				}
 
+				const customValidationHandler = (
```

Comment:
I don't think it makes sense for this `customValidationHandler` to live here alongside other field types. It's only used by the postcode field. If you look at the `ValidatedTextInput` below there is also quite a bit of duplication I feel could be refactored more cleanly. 

How about defining this elsewhere (outside of the `AddressForm` component/utility) and add this to the main component being returned so that `customValidation` is either postcode validation (if the field key is `postcode`) or `undefined` for other types?

Reply:
So like a general validation handler?

Reply:
Just something to improve the convenience because if we move to modal, the local state needs to be used over the data store.